src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs

```diff
@@ -1,138 +1,146 @@
-﻿using System;
+﻿using Autodesk.DesignScript.Runtime;
+using Autodesk.Revit.DB;
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using Autodesk.Revit.DB;
-using Autodesk.DesignScript.Runtime;
-using CoEdge = Autodesk.DesignScript.Geometry.CoEdge;
 using Edge = Autodesk.DesignScript.Geometry.Edge;
 using Face = Autodesk.DesignScript.Geometry.Face;
-using ProtoPt = Autodesk.DesignScript.Geometry.Point;
 using Solid = Autodesk.DesignScript.Geometry.Solid;
 using Surface = Autodesk.DesignScript.Geometry.Surface;
 
 namespace Revit.GeometryConversion
 {
-   [SupressImportIntoVM]
-   public static class DynamoToRevitBRep
-   {
-       /// <summary>
-       /// this method attempts to construct a BRep from a closed solid.
-       /// </summary>
-       /// <param name="sol"></param>
-       /// <param name="performHostUnitConversion"></param>
-       /// <returns></returns>
-       public static GeometryObject ToRevitType(Solid sol,
-          bool performHostUnitConversion = true)
-       {
-            var geometry = ToRevitType(sol, performHostUnitConversion, BRepType.Solid);
-
-           if (geometry == null)
-           {
-               // An unexpected failure occurred when attempting to convert the solid into a Revit BRep.
-               throw new Exception(Properties.Resources.DynamoSolidToRevitBRepFailure);
-           }
+    [SupressImportIntoVM]
+    public static class DynamoToRevitBRep
+    {
+        /// <summary>
+        /// this method attempts to construct a BRep from a closed solid.
+        /// </summary>
+        /// <param name="sol"></param>
+        /// <param name="performHostUnitConversion"></param>
+        /// <param name="materialId"></param>
+        /// <returns></returns>
+        public static GeometryObject ToRevitType(Solid sol,
+          bool performHostUnitConversion = true,
+          ElementId materialId = null)
```

Comment:
Add material id param as an optional param the end in order to maintain backward compability.